Filter function: brightness() (#25057)
mdn/content: files/en-us/web/css/filter-function/brightness/index.md

{{CSSRef}}

> The **`brightness()`** [CSS](/en-US/docs/Web/CSS) {{cssxref("&lt;filter-function&gt;")}} applies a linear multiplier value on the element or input image, making the image appear brighter or darker.

Reviewer: ```suggestion
The **`brightness()`** [CSS](/en-US/docs/Web/CSS) {{cssxref("&lt;filter-function&gt;")}} applies a linear multiplier value on an element or an input image, making the image appear brighter or darker.
```

The following are pairs of equivalent values:

```css
brightness(0) /* Completely black */
brightness(0%)

brightness(0.4) /* 40% brightness, which is 60% darker */
brightness(40%)

brightness(1) /* No effect */
brightness(100%)

brightness(2) /* Double brightness */
brightness(200%)
```

### Formal syntax

{{csssyntax}}

## Examples

### With the backdrop-filter property

This example applies a `brightness()` filter via the `backdrop-filter` CSS property to the paragraph, linearly shifting the colors of area behind the `<p>`.

```css
.container {
  background: url(image.jpg) no-repeat right / contain #d4d5b2;
}
p {
  backdrop-filter: brightness(150%);
  text-shadow: 2px 2px #ffffff;
}
```

```css hidden
.container {
  padding: 5rem 3rem 1rem;
  width: 30rem;
}
p {
  padding: 0.5rem;
  color: #000000;
  font-size: 2rem;
  font-family: sans-serif;
}
```

```html hidden
<div class="container" style="background-image: url(be_fierce.jpg);">
  <p>
    Text on images can be illegible and inaccessible even with a drop shadow.
  </p>
</div>

> ```
> 
> {{EmbedLiveSample('With_the_backdrop-filter_property','100%','280')}}

Reviewer: ```suggestion

#### Result

{{EmbedLiveSample('Applying_brightness_using_the_backdrop_filter_property', '100%', '280')}}

In this example, the colors in the area behind the `<p>` element shift linearly. If the `backdrop-filter` property was set to `brightness(0%)`, the `<div>` area with the `<p>` element would have been black and hidden the image behind. At `brightness(100%)`, the `<div>` area color would be the same as the input `#d4d5b2`, and the image behind would be completely transparent. With the brightness set to `150%` as in this example, the colors in the image behind are getting hidden by the brightness of the `<div>` element.`
```

- How about unhiding the html code? That way readers will be able to view the source `<p>` element on which the property and the function are being applied.
- I've added some explanation after the result to explain what would happen with other brightness values. Feel free to rewrite for technical accuracy.
- I edited the original phrase "linearly shifting the colors of area behind the `<p>`" to "the colors in the area behind the `<p>` element shift linearly". Can this be rephrased for better conveying what's happening, especially only in the part overlapping the image?
- An issue for another time - I did not find documentation for `/ contain` on the [background](https://developer.mozilla.org/en-US/docs/Web/CSS/background) property page although it is used in one of the interactive examples.

Author: `contain` is a value of the `background-size` property. `background` is a shorthand.

> ### With the filter property

Reviewer: ```suggestion
### Applying brightness using the filter property
```

Since this is a simpler example, I would suggest to make it first of the three examples. In the previous one, the reader will have to contend with `brightness` as well as `backdrop-filter` property and that can be tricky.

> This example applies a `brightness()` filter via the `filter` CSS property linearly color shifting the entire element, including content, border, and background image.

Reviewer: ```suggestion
In this example, a `brightness()` filter is applied to the entire element, including content, border, and background image via the [`filter`](/en-US/docs/Web/CSS/filter) CSS property. The result shows three variations of different brightness values.
```

```css
p:first-of-type {
  filter: brightness(50%);
}
p:last-of-type {

>   filter: contrast(200%);

Reviewer: ```suggestion
  filter: brightness(200%);
```

}
```

```css hidden
p {
  text-shadow: 2px 2px blue;
  background-color: magenta;
  color: palegoldenrod;
  border: 1em solid rebeccapurple;
  box-shadow: inset -5px -5px red, 5px 5px yellow;
  padding: 0.25rem;
  font-size: 1.25rem;
  font-family: sans-serif;
  width: 85vw;
}
```

```html hidden
<p>This paragraph has reduced brightness.</p>
<p>This paragraph has normal brightness.</p>
<p>This paragraph has increased brightness.</p>
```

> {{EmbedLiveSample('With_the_filter_property','100%','280')}}

Reviewer: because of my edit suggestion to the title...

```suggestion
{{EmbedLiveSample('Applying_brightness_using_the_filter_property','100%','280')}}
```

> ### With url() SVG brightness filter

Reviewer: ```suggestion
### Applying brightness using the url() SVG brightness filter
```

> The SVG {{SVGElement("filter")}} element is used to define custom filter effects that can then be referenced by {{htmlattrxref("id")}}. The `<filter>`'s {{SVGElement("feComponentTransfer")}} primitive enables pixel-level color remapping. To create a filter that darkens the content on which it is applied by 25%, we set the `slope` attribute to 0.75, for 75% of the original brightness. We can then reference the filter by `id`.

Reviewer: ```suggestion
The SVG {{SVGElement("filter")}} element is used to define custom filter effects that can then be referenced by {{htmlattrxref("id")}}. The `<filter>` element's {{SVGElement("feComponentTransfer")}} primitive enables pixel-level color remapping.

In this example, to create a filter that darkens the content on which it is applied by 25% (i.e., 75% of the original brightness), the `slope` attribute is set to `0.75`. We can then reference the filter by `id`.
```

Given the following:

```css hidden
.filter {
  filter: brightness(0.75);
}
svg {
  position: absolute;
}
```

```html
<svg role="none">
  <filter id="darken25" color-interpolation-filters="sRGB">
    <feComponentTransfer>
      <feFuncR type="linear" slope="0.75" />
      <feFuncG type="linear" slope="0.75" />
      <feFuncB type="linear" slope="0.75" />
    </feComponentTransfer>
  </filter>
</svg>
```

The following declarations produce similar effects:

```css
filter: brightness(75%);
filter: url(#darken25); /* with embedded SVG */
filter: url(folder/fileName.svg#darken25); /* external svg filter definition */
```

> This example shows three images: the image with a `brightness()` filter function applied, the image with a similar SVG brightness function applied, and the original image for comparison:

Reviewer: ```suggestion
In the images below, the first one has a `brightness()` filter function applied, the second one has a similar SVG brightness function applied, and the third is the original image for comparison.
```

The following are pairs of equivalent values:

```css
brightness(0) /* Brightness is reduced to zero, so input turns black */
brightness(0%)

brightness(0.4) /* Brightness of input is reduced to 40%, so input is 60% darker */
brightness(40%)

brightness(1) /* Brightness of input is not changed */
brightness(100%)

brightness(2) /* Brightness of input is doubled */
brightness(200%)

### Formal syntax

{{csssyntax}}

## Examples

### Applying brightness using the backdrop-filter property

This example shows how to apply the `brightness()` filter to a paragraph via the [`backdrop-filter`](/en-US/docs/Web/CSS/backdrop-filter) CSS property.

#### CSS

```css
.container {
  background: url(image.jpg) no-repeat right / contain #d4d5b2;
}
p {
  backdrop-filter: brightness(150%);
  text-shadow: 2px 2px #ffffff;
}
```

```css hidden
.container {
  padding: 5rem 3rem 1rem;
  width: 30rem;
}
p {
  padding: 0.5rem;
  color: #000000;
  font-size: 2rem;
  font-family: sans-serif;
}
```

```html hidden
<div class="container" style="background-image: url(be_fierce.jpg);">
  <p>
    Text on images can be illegible and inaccessible even with a drop shadow.
  </p>

> </div>

Reviewer: the closing code fence for this html markdown is missing

## See also

> - [CSS filter effects](/en-US/docs/Web/CSS/filter_effects/) module

Author: ```suggestion
- [CSS filter effects](/en-US/docs/Web/CSS/filter_effects) module
```

- The other {{cssxref("&lt;filter-function&gt;")}} functions available to be used in values of the {{cssxref("filter")}} and {{cssxref("backdrop-filter")}} properties include:
  - {{cssxref("filter-function/blur", "blur()")}}
  - {{cssxref("filter-function/contrast", "contrast()")}}
  - {{cssxref("filter-function/drop-shadow", "drop-shadow()")}}
  - {{cssxref("filter-function/grayscale", "grayscale()")}}
  - {{cssxref("filter-function/hue-rotate", "hue-rotate()")}}
  - {{cssxref("filter-function/invert", "invert()")}}
  - {{cssxref("filter-function/opacity", "opacity()")}}
  - {{cssxref("filter-function/saturate", "saturate()")}}

>   - {{cssxref("filter-function/sepia", "sepia()")}}

Reviewer: Not touching this "See also" yet. But leaning towards - listing the filter-functions in the same list item and moving the module link to the end.
